File: launch_school/Programming_Foundations/small_problems/double_char.rb
# def repeater(str)
#   new_str = ''
#   str.chars.each {|char| new_str << char << char}
#   new_str
# end

# p repeater('Hello') == "HHeelllloo"
# p repeater("Good job!") == "GGoooodd  jjoobb!!"
# p repeater('') == ''

ALPHABETS = ('A'..'Z').to_a
VOWELS = %W(A E I O U)
CONSONANTS = ALPHABETS - VOWELS

def double_consonants(str)
  new_str = ''
  str.chars.each do |char|
      new_str << char 
      new_str << char if CONSONANTS.include?(char.upcase)
  end
  new_str
end

p double_consonants('String') == "SSttrrinngg"
p double_consonants("Hello-World!") == "HHellllo-WWorrlldd!"
p double_consonants("July 4th") == "JJullyy 4tthh"
p double_consonants('') == ""

File: launch_school/JS210-Fundamentals-of-Javascript/small_problems/easy5/double_char.js
'use strict';

let repeater = function (string) {
  return string.split('').map(char => char.repeat(2)).join('');
};

console.log(repeater('Hello'));        // "HHeelllloo"
console.log(repeater('Good job!'));    // "GGoooodd  jjoobb!!"
console.log(repeater(''));             // ""
//

let doubleConsonants = function (string) {
  const doubleOnlyConsonant =  (char) => {
    if (/[AEIOUaeiou]/.test(char)) {
      return char;
    } else if (/[A-Za-z]/.test(char)) {
      return char.repeat(2);
    } else {
      return char;
    }
  };

  return [...string].map(doubleOnlyConsonant).join('');
};

console.log(doubleConsonants('String'));          // "SSttrrinngg"
console.log(doubleConsonants('Hello-World!'));    // "HHellllo-WWorrlldd!"
console.log(doubleConsonants('July 4th'));        // "JJullyy 4tthh"
console.log(doubleConsonants(''));                // ""
